### ISSUE TITLE
Lock digests not always removed after forced sidekiq worker shutdown

### ISSUE THREAD

**Issue by KieranP** [mhenrixon/sidekiq-unique-jobs#861]
**Describe the bug**

We use Sidekiq Pro's super_fetch so that when a worker is forceably shut down (usually due to a deploy or runaway RAM usage), the job is then added back to the enqueued list when sidekiq starts back up. This works well 99% of the time.

But occasionally what we are seeing is that the digest locks for jobs running when the worker is shut down remain in place, and so Sidekiq fails to reenqueue the job when a replacement worker is spun up, and new jobs of the same type won't queue because the lock remains in place from before the shutdown.

This results in a stuck state. We have to manually go in and delete the locks in order for the jobs to be able to requeue.

**Expected behavior**

When a Sidekiq worker restarts, before sidekiqs super_fetch kicks in, sidekiq-unique-jobs should remove any locks that were linked to jobs on workers that no longer exist.

e.g. pseudo ruby code: `SidekiqUniqueLocks.where.not(worker_id: Sidekiq.workers.map(&:id)).delete_all`

**Current behavior**

Seems it removes the locks most of the time but it has a glitch where it doesn't, and it happens often enough to stop our background jobs from running smoothly.

**Worker class**

```ruby
class MyWorker
  include Sidekiq::Worker
  sidekiq_options(
    queue: :default,
    lock: :until_executed,
  )
end
```

**Additional context**

**Comment by glennfu** [comment]
This is still an issue in v8.0.11. If the process dies, the reaper will not clean it up and requires manual intervention. I considered a `lock_ttl` as a crude fallback but as this sets at enqueue time and not `job started running` time I can't be sure of a reasonable number that won't make things worse if the queue is backed up under heavy load.

**Comment by chuan29812** [comment]
yup, this is still a problem. Happens the most during deployment, and sometimes triggers out of nowhere. Have to go into the dashboard to do delete_all during off peak hour to make sure new jobs with the same digest can be run, thank god we dont have anything crucial that could be blocked by having the same digest being left around

**Comment by mhenrixon** [comment]
The reaper's `active?` check handles this scenario — once a Sidekiq process is forcefully terminated, it's removed from the `processes` set (after the heartbeat timeout). At that point, its locks no longer have an active process and become orphans, which the reaper cleans up in the next cycle. Our #890 fix (PR #931) also helps: `until_executed` now preserves the lock on failure/shutdown, which gives Sidekiq Pro's `super_fetch\!` a chance to re-enqueue the job while the lock is still held.
